flocker/ca/_ca.py

```diff
@@ -445,9 +461,18 @@ def initialize(cls, path, authority, begin=None, uuid=None):
             path=path, keypair=keypair, certificate=cert)
         credential.write_credential_files(
             key_filename, cert_filename)
-        instance = cls(credential=credential, uuid=uuid)
+        instance = cls(credential=credential)
         return instance
 
+    @property
+    def uuid(self):
+        # We need to strip the "node-" prefix:
```

Comment:
If we are going to strip the prefix, somewhere we should verify that that is in fact waht we are stripping.